modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc

```diff
@@ -1339,6 +1339,8 @@ bool PathBoundsDecider::CheckLaneBoundaryType(
     lane_boundary_type = hdmap::LeftBoundaryType(waypoint);
   } else if (lane_borrow_info == LaneBorrowInfo::RIGHT_BORROW) {
     lane_boundary_type = hdmap::RightBoundaryType(waypoint);
+  } else {
```

Comment:
if you want to check NO_BORROW, can u please move this to the beginning of this function, for better performance? thx

Reply:
Is this function in a hot loop? It seems like these are mostly quick checks.

If we want to have minimal latency for the `lane_borrow_info==LaneBorrowInfo::NO_BORROW` case, another option is doing the check before the `CheckLaneBoundaryType` call in https://github.com/ApolloAuto/apollo/blob/15f375a8ff1464bdab9eb65cac091c72ab7a5613/modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc#L785. It's probably good to have the `CheckLaneBoundaryType` method be valid for all values of `lane_borrow_info`.

Both?

Reply:
yes. you can add this check at both places.